### PR TITLE
Add vrs recipe

### PR DESCRIPTION
**Pull request by SeaOtocinclus**
Adding a recipe to build and package vrs (a library and tools to manipulate VRS files)

Looking for help on review and make this to build 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source. 
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by SeaOtocinclus** [comment]
@conda-forge/help-c-cpp

Hello, I'm seeking for some guidance.
I'm trying to add a new recipe for VRS and here is my status:
Linux, Windows (Seems ok):
- seems ok but perfectible, not sure I'm doing the best with some libraries names that are duplicated (maybe?)
macOS (need help):
- I have no idea why link is not working on mac (code is build but not link, symbols are not found) - On my local machine I can build fine with a MAC and a conda environment set by PIXI
 https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1027757&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=1909
windows (need help):

**Comment by SeaOtocinclus** [comment]
cc. @traversaro FYI I'm trying to continue to extend how people can reach VRS and its eco-system. Maybe you will help me improve this recipe or detect what is going wrong with mac...

**Comment by traversaro** [comment]
> cc. @traversaro FYI I'm trying to continue to extend how people can reach VRS and its eco-system. Maybe you will help me improve this recipe or detect what is going wrong with mac...

I just checked the first error:

~~~
FAILED: vrs/utils/cli/libvrs_utils_cli.dylib 
: && $BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang++ -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/vrs-1.2.5 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -D_LIBCPP_DISABLE_AVAILABILITY -O3 -DNDEBUG -isysroot /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.15 -dynamiclib -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,$PREFIX/lib -L$PREFIX/lib -o vrs/utils/cli/libvrs_utils_cli.dylib -install_name @rpath/libvrs_utils_cli.dylib vrs/utils/cli/CMakeFiles/vrs_utils_cli.dir/CliParsing.cpp.o vrs/utils/cli/CMakeFiles/vrs_utils_cli.dir/CompressionBenchmark.cpp.o vrs/utils/cli/CMakeFiles/vrs_utils_cli.dir/DataExtraction.cpp.o vrs/utils/cli/CMakeFiles/vrs_utils_cli.dir/ListRecords.cpp.o vrs/utils/cli/CMakeFiles/vrs_utils_cli.dir/MakeZeroFilterCopier.cpp.o vrs/utils/cli/CMakeFiles/vrs_utils_cli.dir/PrintRecordFormatRecords.cpp.o vrs/utils/cli/CMakeFiles/vrs_utils_cli.dir/PrintRecordFormats.cpp.o  -Wl,-rpath,$SRC_DIR/build/vrs/utils -Wl,-rpath,$SRC_DIR/build/vrs/helpers -Wl,-rpath,$SRC_DIR/build/vrs/os -Wl,-rpath,$SRC_DIR/build/vrs/oss/logging  vrs/utils/libvrs_utils.dylib  vrs/helpers/libvrs_helpers.dylib  vrs/os/libvrs_os.dylib  vrs/oss/logging/libvrs_logging.dylib  $PREFIX/lib/libfmt.10.2.1.dylib && :
Undefined symbols for architecture x86_64:
  "vrs::Recordable::~Recordable()", referenced from:
      vrs::utils::(anonymous namespace)::ZeroFilter::~ZeroFilter() in MakeZeroFilterCopier.cpp.o
      vrs::utils::(anonymous namespace)::ZeroFilter::~ZeroFilter() in MakeZeroFilterCopier.cpp.o
~~~

Are you sure that the linking is done correctly at the CMake level? At a first glance `ZeroFilter` definition in https://github.com/SeaOtocinclus/vrs/blob/1.2.5/vrs/utils/cli/MakeZeroFilterCopier.cpp#L25 uses `RecordFilterCopier` from `vrs_utils` https://github.com/SeaOtocinclus/vrs/blob/1.2.5/vrs/utils/FilterCopyHelpers.h#L181, that itself uses `RecordFormatStreamPlayer` from the `vrslib` lib (see https://github.com/SeaOtocinclus/vrs/blob/1.2.5/vrs/RecordFormatStreamPlayer.h#L60), but `vrslib` is not a public dependency of `vrs_utils`, see https://github.com/SeaOtocinclus/vrs/blob/1.2.5/vrs/utils/CMakeLists.txt#L27-L38 . 

I am afraid that there may several errors like that in the codebase, that perhaps are hidden by the default settings in linker in different versions/operating systems. To detect such problems also on ld's in Linux, you can use the `-Wl,--unresolved-symbols=report-all` option, see for example https://github.com/robotology/wb-toolbox/blob/19b7408d7f2ba69ba82ced58d3a103015edd0309/CMakeLists.txt#L71-L77 .

**Comment by SeaOtocinclus** [comment]
Thank you @traversaro , it seems the error only shows on CLANG but not GCC. As you said it seems to be missing dependencies and CMake modules should be better defined.
- I will try first to see if the build pass as static libs for the VRS libraries
- Then try to see if I can modify the CMake rules/modules of VRS to avoid such linking issues

**Comment by traversaro** [comment]
> Thank you @traversaro , it seems the error only shows on CLANG but not GCC.

Did you tried with Clang on Linux? Because I expect the difference to be the default behavior of linkers in macOs and Linux, not of compilers.

**Comment by SeaOtocinclus** [comment]
As predicted, static build for VRS libraries (but still deps in dynamic libs from CONDA are working).
I will see if I can modify VRS or not to fix this DYNAMIC LIB build behavior, else I will rely on first shipping a static build for CONDA, and update later to Dynamic build.

**Comment by SeaOtocinclus** [comment]
@conda-forge/help-c-cpp, ready for review!

Looking forward review, thank you for your feedback and help!

**Comment by SeaOtocinclus** [comment]
@xhochy any information somewhere on how long we need to wait to receive reviews? (sorry for the personal ping)

File: recipes/vrs/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - mesalib  # [linux]
+    # dependencies to be added for RUN step
+    - libpng
+    - lz4-c
+    - zstd
+    - libboost
+
+test:
+  commands:
+    - test -f $PREFIX/bin/vrsplayer  # [not win]
```
Comment (xhochy) [inline recipes/vrs/meta.yaml:42]: Can you please add tests for Windows, too?

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/vrs/meta.yaml (hunk 1 above)
Comment (SeaOtocinclus) [reply recipes/vrs/meta.yaml:42]: Do you know if I can use the same command line to know if the file is present?
I.e. is the default path different for windows?

Comment (xhochy) [reply recipes/vrs/meta.yaml:42]: Probably the following is what you want:
`if not exist %LIBRARY_PREFIX%\bin\vrsplayer.exe exit 1`

Comment (SeaOtocinclus) [reply recipes/vrs/meta.yaml:42]: @xhochy Thank you, your suggestion worked.

**Review by xhochy — APPROVED** [review]

**Comment by SeaOtocinclus** [comment]
Thank you @xhochy for jumping on reviewing my PR.

**Pull request merged**